src/Cli/dotnet/Commands/Test/SolutionAndProjectUtility.cs

```diff
@@ -230,23 +229,38 @@ public static IEnumerable<ParallelizableTestModuleGroupWithSequentialInnerModule
         }
 
         string targetFramework = project.GetPropertyValue(ProjectProperties.TargetFramework);
-        RunProperties runProperties = GetRunProperties(project, loggers);
+		string projectFullPath = project.GetPropertyValue(ProjectProperties.ProjectFullPath);
 
-        // dotnet run throws the same if RunCommand is null or empty.
-        // In dotnet test, we are additionally checking that RunCommand is not dll.
-        // In any "default" scenario, RunCommand is never dll.
-        // If we found it to be dll, that is user explicitly setting RunCommand incorrectly.
-        if (string.IsNullOrEmpty(runProperties.Command) || runProperties.Command.HasExtension(CliConstants.DLLExtension))
+
+        // Only get run properties if IsTestingPlatformApplication is true
+        RunProperties runProperties;
+        if (isTestingPlatformApplication)
         {
-            throw new GracefulException(
-                string.Format(
-                    CliCommandStrings.RunCommandExceptionUnableToRun,
-                    "dotnet test",
-                    "OutputType",
-                    project.GetPropertyValue("OutputType")));
-        }
+            runProperties = GetRunProperties(project, loggers);
 
-        string projectFullPath = project.GetPropertyValue(ProjectProperties.ProjectFullPath);
+            // dotnet run throws the same if RunCommand is null or empty.
+            // In dotnet test, we are additionally checking that RunCommand is not dll.
+            // In any "default" scenario, RunCommand is never dll.
+            // If we found it to be dll, that is user explicitly setting RunCommand incorrectly.
+            if (string.IsNullOrEmpty(runProperties.Command) || runProperties.Command.HasExtension(CliConstants.DLLExtension))
+            {
+                throw new GracefulException(
+                    string.Format(
+                        CliCommandStrings.TestCommandExceptionUnableToRun,
+						projectFullPath,
+						"dotnet test",
+                        "OutputType",
+                        project.GetPropertyValue("OutputType")));
+            }
+        }
+        else
+        {
+            // For VSTest test projects, create minimal RunProperties
+            runProperties = new RunProperties(
+                project.GetPropertyValue(ProjectProperties.TargetPath),
+                null,
+                null);
```

Comment:
Can't we simply do `runProperties = null;`? We should never read anything from RunProperties when it's not MTP.